src/librustc_metadata/astencode.rs

```diff
@@ -113,8 +72,13 @@ pub fn decode_inlined_item<'a, 'tcx>(cdata: &cstore::CrateMetadata,
                                      orig_did: DefId)
                                      -> &'tcx InlinedItem {
     debug!("> Decoding inlined fn: {:?}", tcx.item_path_str(orig_did));
-    let id_range_doc = ast_doc.get(c::tag_id_range);
-    let from_id_range = IdRange::decode(&mut id_range_doc.opaque()).unwrap();
+    let from_id_range = {
+        let decoder = &mut ast_doc.get(c::tag_id_range).opaque();
```

Comment:
We could use a different encoder here, right? Like, a newtype? (In order to signal we don't want translation)

But I think you revisit this later...

NBD, anyhow.